application/libraries/Api/Command/V1/SurveyPatch/OpHandlerQuestionCondition.php

```diff
@@ -0,0 +1,944 @@
+     *                             "csrctoken": "{TOKEN:LASTNAME}",
+     *                             "ConditionConst": "",
+     *                             "prevQuestionSGQA": "@453614X608X15978@",
+     *                             "ConditionRegexp": "",
+     *                             "cqid": 0,
+     *                             "canswersToSelect": "",
+     *                             "editSourceTab": "#SRCTOKENATTRS",
+     *                             "editTargetTab": "#PREVQUESTIONS"
+     *                         }
+     *                     ]
+     *                 }
+     *             ]
+     *         }}
+     *     ]
+     * }
+     * updateCondition (field-regex)
+     * {
+     *     "patch": [{
+     *         "op": "update",
+     *         "entity": "questionCondition",
+     *         "error": false,
+     *         "props": {
+     *             "qid": 15977,
+     *             "scenarios": [
+     *                 {
+     *                     "scid": 123,
+     *                     "conditions": [
+     *                         {
+     *                             "cid": 2601,
+     *                             "action": "updateCondition",
+     *                             "method": "RX",
+     *                             "csrctoken": "{TOKEN:LASTNAME}",
+     *                             "ConditionConst": "",
+     *                             "ConditionRegexp": "abc",
+     *                             "cqid": 0,
+     *                             "canswersToSelect": "",
+     *                             "editSourceTab": "#SRCTOKENATTRS",
+     *                             "editTargetTab": "#REGEXP"
+     *                         }
+     *                     ]
+     *                 }
+     *             ]
+     *         }}
+     *     ]
+     * }
+     * deleteCondition
+     * {
+     *     "patch": [{
+     *         "op": "delete",
+     *         "entity": "questionCondition",
+     *         "error": false,
+     *         "props": {
+     *             "qid": 15977,
+     *             "scenarios": [
+     *                 {
+     *                     "scid": 123,
+     *                     "conditions": [
+     *                         {
+     *                             "cid": 2601,
+     *                             "action": "deleteCondition"
+     *                         }
+     *                     ]
+     *                 }
+     *             ]
+     *         }}
+     *     ]
+     * }
+     *
+     *
+     * @param OpInterface $op
+     * @return void
+     * @throws OpHandlerException
+     * @throws PersistErrorException
+     * @throws NotFoundException
+     * @throws PermissionDeniedException
+     */
+    public function handle(OpInterface $op): void
+    {
+        $qid = $op->getProps()['qid'] ?? '';
+        if (isset($op->getProps()['action'])) {
+            $action = $op->getProps()['action'];
+            switch ($action) {
```

Comment:
We would create a separate op handler for each action. The only exception to this right now is the survey patch endpoint. Maybe it makes sense to split each action out into a separate op handler?

Reply:
@kevin-foster-uk to me it seems clearer if we are able to see the condition actions in a class dedicated for conditions. If we created separate op handlers for the 10+ actions, that would seem confusing to me. But if it is something important for our conventions, then we can of course do it. Would we need to refactor this Op Handler and cut it into small pieces separated into different Op Handlers?